### ISSUE TITLE
Custom iterators not supported in for-of loops

### ISSUE THREAD

**Issue by infinnie** [Microsoft/TypeScript#24314]
**TypeScript Version:**  2.9.0 rc

**Search Terms:**

> for of iterator

**Code**

```ts
var Thing = {
    [Symbol.iterator]: function () {
        return {
            next: function () {
                return {
                    done: false,
                    value: 1
                };
            }
        };
    }
};

for (var a of Thing) {
    console.log(a);
    break;
}
```

**Expected behavior:** No error is reported in the editor and during compilation, and 1 would be logged if executed

**Actual behavior:** The editor in the Playground reports an error:
> Type '{ [Symbol.iterator]: () => { next: () => { done: boolean; value: number; }; }; }' is not an array type or a string type.

Visual Studio 2015 Update 3 with TypeScript 2.9 tools installed also reports an error at the place of [Symbol.iterator].

**Playground Link:** https://www.typescriptlang.org/play/#src=var%20Thing%20%3D%20%7B%0D%0A%20%20%20%20%5BSymbol.iterator%5D%3A%20function%20()%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20next%3A%20function%20()%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20return%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20done%3A%20false%2C%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20value%3A%201%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%3B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0D%0A%20%20%20%20%20%20%20%20%7D%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%3B%0D%0A%0D%0Afor%20(var%20a%20of%20Thing)%20%7B%0D%0A%20%20%20%20console.log(a)%3B%0D%0A%20%20%20%20break%3B%0D%0A%7D
**Related Issues:**

**Comment by infinnie** [comment]
OK, --downlevelIteration
